packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -1871,34 +2366,15 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin {
       _debugLocked = true;
       return true;
     }());
-    final Route<dynamic> oldRoute = _history.last;
+    final Route<dynamic> oldRoute = _history.last.route;
```

Comment:
The last route may already be on its way out and hence cannot be replaced. Do we need to get the last present route?

Reply:
Also, what's the oldRoute variable for? Doesn't seem to be used below?